server/sonar-web/src/main/js/components/SourceViewer/components/MeasuresOverlay.tsx

```diff
@@ -0,0 +1,474 @@
+          </div>
+        </div>
+      </div>
+    );
+  };
+
+  renderAllMeasures = () => {
+    const domains = groupBy(Object.values(this.state.measures), measure => measure.metric.domain);
+    const domainKeys = Object.keys(domains);
+    const odd = domainKeys.filter((_, index) => index % 2 === 1);
+    const even = domainKeys.filter((_, index) => index % 2 === 0);
+    return (
+      <div className="source-viewer-measures source-viewer-measures-secondary js-all-measures">
+        <div className="source-viewer-measures-section source-viewer-measures-section-big">
+          {odd.map(domain => this.renderDomain(domain, domains[domain]))}
+        </div>
+        <div className="source-viewer-measures-section source-viewer-measures-section-big">
+          {even.map(domain => this.renderDomain(domain, domains[domain]))}
+        </div>
+      </div>
+    );
+  };
+
+  render() {
+    const { branch, component } = this.props;
+    const { loading } = this.state;
+
+    return (
+      <Modal contentLabel="" large={true} onRequestClose={this.props.onClose}>
+        <div className="modal-container source-viewer-measures-modal">
+          <div className="source-viewer-header-component source-viewer-measures-component">
+            <div className="source-viewer-header-component-project">
+              <QualifierIcon className="little-spacer-right" qualifier="TRK" />
+              <Link to={getProjectUrl(component.project, branch)}>{component.projectName}</Link>
+
+              {component.subProject && (
+                <>
+                  <QualifierIcon className="big-spacer-left little-spacer-right" qualifier="BRC" />
+                  <Link to={getProjectUrl(component.subProject, branch)}>
+                    {component.subProjectName}
+                  </Link>
+                </>
+              )}
+            </div>
+
+            <div className="source-viewer-header-component-name">
+              <QualifierIcon className="little-spacer-right" qualifier={component.q} />
+              {component.path || component.longName}
+            </div>
+          </div>
+
+          {loading ? (
+            <i className="spinner" />
+          ) : (
+            <>
+              {component.q === 'UTS' ? (
+                <>
+                  {this.renderTests()}
+                  <MeasuresOverlayTestCases branch={branch} componentKey={component.key} />
+                </>
+              ) : (
+                <div className="source-viewer-measures">
+                  {this.renderLines()}
+                  {this.renderIssues()}
+                  {this.renderCoverage()}
+                  {this.renderDuplications()}
+                </div>
+              )}
+            </>
+          )}
+
+          <div className="spacer-top">
+            {this.state.showAllMeasures ? (
+              this.renderAllMeasures()
+            ) : (
+              <a className="js-show-all-measures" href="#" onClick={this.handleAllMeasuresClick}>
```

Comment:
I think this should be a `button` with `.button-link`

Reply:
I will do it after I merge my another PR 😉